guide/src/configuration.md

```diff
@@ -0,0 +1,89 @@
+# Configuring Clippy
```

Comment:
This should be at the end of the configuration doc. Also I'd like to rewrite this section a bit (noting that this config file was never stabilized).

Reply:
> This should be at the end of the configuration doc. 

Not exactly sure what you mean here. 

> Also I'd like to rewrite this section a bit (noting that this config file was never stabilized).

Yeah, I imagine a lot of stuff can use some cleanup and rewriting. I'm happy to help with that as well.